src/main/java/com/tananaev/adblib/AdbConnection.java

```diff
@@ -301,7 +338,10 @@ public AdbStream open(String destination) throws UnsupportedEncodingException, I
                 wait();
 
             if (!connected) {
-                throw new IOException("Connection failed");
+                if (authorisationFailed)
```

Comment:
Should we have a helper method for this block of code. It looks like it's duplicated in several places.

Reply:
Good idea, I've just committed 2ac631c with this change